src/puppetlabs/puppetdb/facts.clj

```diff
@@ -30,6 +30,15 @@
    (s/one s/Str "version")
    (s/one s/Str "provider")])
 
+(def hashed-package-tuple
+  (conj package-tuple
+        (s/one s/Str "package_hash")))
+
+(defn package-tuple-name [package-tuple] (nth package-tuple 0))
+(defn package-tuple-version [package-tuple] (nth package-tuple 1))
+(defn package-tuple-provider [package-tuple] (nth package-tuple 2))
+(defn package-tuple-hash [hashed-package-tuple] (nth hashed-package-tuple 3))
```

Comment:
the last one is used. I put the first ones there in a fit of sicp-fueled encapsulation.